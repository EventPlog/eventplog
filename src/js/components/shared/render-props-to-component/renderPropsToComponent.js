import React, { Component } from 'react'
import Error from 'js/components/shared/loading/Error'


class ComponentWithContainerProps extends Component {
  state = { hasError: false, msg: '' }

  componentDidCatch(error, info) {
    this.setState({ hasError: true });
    // You can also log the error to an error reporting service
    // logErrorToMyService(error, info);
    console.log(error, info)
  }

  render() {
    if (this.state.hasError) {
      return <Error msg={this.state.msg} />
    }

    const {container: Container, component: Component} = this.props
    return (
      <Container>
        {(props) => <Component {...props} />}
      </Container>
    )
  }
}

const renderComponentWithProps = (container, component) => () => (
  <ComponentWithContainerProps {...{container, component}} />
)

export default renderComponentWithProps