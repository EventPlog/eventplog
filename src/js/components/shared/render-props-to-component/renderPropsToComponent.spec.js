import React from 'react';
import renderPropsToComponent  from './renderPropsToComponent';
import { shallow, mount } from 'enzyme';

describe('renderPropsToComponent', () => {
  const containerProp = {name: 'John Doe'}
  const Container = (props) => <div>{props.children(containerProp)}</div>
  const ChildComponent = (props) => <p>{props.name}</p>
  let wrapper;

  beforeAll(() => {
    wrapper = mount( renderPropsToComponent(Container, ChildComponent) );
  })

  test('it should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(Container).length).toEqual(1);
    expect(wrapper.find(ChildComponent).length).toEqual(1);
  })

  test('it should pass the container prop to the child', () => {
    expect(wrapper.find('p').text()).toEqual(containerProp.name)
  })
});