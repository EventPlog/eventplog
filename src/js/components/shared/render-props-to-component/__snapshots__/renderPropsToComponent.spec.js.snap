// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderPropsToComponent it should render correctly 1`] = `
<Component>
  <ComponentWithContainerProps
    component={[Function]}
    container={[Function]}
  >
    <Container>
      <div>
        <ChildComponent
          name="John Doe"
        >
          <p>
            John Doe
          </p>
        </ChildComponent>
      </div>
    </Container>
  </ComponentWithContainerProps>
</Component>
`;
